[Gerald Ng Jun Xiang] iP

geraldngjx opened the pull request:
# GeraldBot: Your Ultimate Task Manager

> "Meet your deadlines, use GeraldBot today." – Gerald Ng

GeraldBot is your go-to solution for managing tasks efficiently. It's designed to simplify your task management process and boost your productivity. Here's why you'll love it:

- **User-Friendly**: GeraldBot is easy to navigate and doesn't overwhelm you with complex features.
- **Speedy**: It's lightning-fast, making task management a breeze.
- **Intuitive**: Quickly add and organize tasks with a few clicks.
- **Flexible**: Tailor it to your needs, whether it's for work, study, or personal life.

## Getting Started

1. **Download**: Get GeraldBot [here](https://github.com/geraldngjx/ip). Then, clone the project.
2. **Installation**: Follow the ReadMe to set up GeraldBot.
3. **Add Your Tasks**: Start adding your tasks and let GeraldBot take care of the rest.
4. **Stay Organized**: Enjoy a clutter-free task list and stay on top of your priorities.
5. **It's FREE**: All these features are available to you at no cost.

## Key Features

- **Task Management**: Easily add, edit, and mark tasks as complete.
- **Deadlines and Reminders** (coming soon - hopefully): Never miss an important date or task.
- **Customization** (coming soon - hopefully): Personalize GeraldBot to match your workflow.

Whether you're a student, professional, or anyone in need of effective task management, GeraldBot has you covered.

_Embrace efficient task management with GeraldBot, your new favorite productivity tool._

---

**Note**: This PR demonstrates the use of GitHub Flavored Markdown (GFMD) for clear and engaging communication.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

geraldngjx committed:
Implement feature to greet user then exit and rename the bot

geraldngjx committed:
Implement feature to echo commands by user

geraldngjx committed:
Add feature to store text by user and display storage when requested

geraldngjx committed:
Add feature to mark and unmark task as done

geraldngjx committed:
Add support for tracking three types of tasks

geraldngjx committed:
Add automated text ui testing

geraldngjx committed:
Add error handling to handle incorrect inputs by user

geraldngjx committed:
Update automated text ui testing to reflect new changes

geraldngjx committed:
Add feature to delete task from list

geraldngjx committed:
Update automated text ui testing to reflect new changes for delete feature

geraldngjx committed:
Fix style to remove unnecessary line breaks

geraldngjx committed:
Update automated text ui testing to reflect new changes due to removing of unnecessary line breaks

geraldngjx committed:
Add java enums to classify commands by TaskType

geraldngjx committed:
Add feature to save task list to storage and add file formatting methods to task types

geraldngjx committed:
Fix formatting bugs that prevent data from being read from the storage

geraldngjx committed:
Add features for file marking and deletion in the storage

geraldngjx committed:
Merge branch 'branch-Level-7'

geraldngjx committed:
Add feature to understand date and time for Deadline Task

geraldngjx committed:
Merge branch 'branch-Level-8'

geraldngjx committed:
Refactor code to extract Ui, Parser, TaskList class

geraldngjx committed:
Merge branch 'branch-A-MoreOOP'

geraldngjx committed:
Remove unnecessary display line commands and extracted common methods into Ui

geraldngjx committed:
Merge branch 'branch-A-MoreOOP'

geraldngjx committed:
Organise the classes into suitable java packages

geraldngjx committed:
Merge branch 'master' into add-gradle-support

geraldngjx committed:
Add a few JUnit tests to test functionality of Storage and Parser

geraldngjx committed:
Edit build.gradle to create jar file

geraldngjx committed:
Add javadocs to the code

geraldngjx committed:
Add feature to find a task by searching for a keyword

geraldngjx committed:
Fix merge conflict due to merging of branch-Level-9 to master

geraldngjx committed:
Fix style issues according to checkstyle tool

geraldngjx committed:
Add GUI container for chatbot

geraldngjx committed:
Add dialog boxes

geraldngjx committed:
Add custom behaviour to DialogBox

geraldngjx committed:
Add images to the repository

geraldngjx committed:
Add fxml to the application

geraldngjx committed:
Fix bug where GeraldBot is unable to parse input correctly

geraldngjx committed:
Merge branch 'branch-Level-10'

geraldngjx committed:
Fix styling issues based on CheckStyle tool

geraldngjx committed:
Fix bug where bye command cannot be parsed

lsyurea commented on src/main/java/geraldbot/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+package geraldbot;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
```
I like that your imports are alphabetically arranged

lsyurea commented on src/main/java/geraldbot/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
```
I like that you used Flips which is the correct way to write your javadocs

lsyurea commented on src/main/java/geraldbot/Launcher.java:
```diff
@@ -0,0 +1,12 @@
+
+import javafx.application.Application;
+
+/**
+ * A launcher class to workaround classpath issues.
+ */
```
Nothing wrong with this javaDocs as well. Personally, using Constructs a launcher class ... sounds nicer

lsyurea commented on src/main/java/geraldbot/exception/DukeException.java:
```diff
@@ -0,0 +1,16 @@
+
+    /**
+     * Constructs a DukeException with the specified error message.
+     *
```
Should there be a space in javadocs?

lsyurea reviewed (commented):
Overall, this codebase seem clean and better than mine :)

geraldngjx committed:
Refactor Parser.java to adhere to coding standards

geraldngjx committed:
Add JavaDocs to new Parser.java functions

geraldngjx committed:
Add java assertion to document important assumptions in the code

The code lacks java assertions, making it challenging to document
assumptions which are critical at various points in the code.

Adding java assertions is crucial for documenting assumptions. It
helps identify and also prevent unexpected issues during
development and runtime.

This commit introduces java assertions at key points in the codebase.

This enhances code clarity and maintainability.

geraldngjx committed:
Merge pull request #1 from geraldngjx/branch-A-Assertions

Add java assertion to document important assumptions in the code

geraldngjx committed:
Refactor code to enhance code quality

Code has some inaccurate naming which could be misleading.
Changed existing variable names to make it more accurate.

Changes made:
* Renamed the variable "lst" to "taskList" to provide a more descriptive and intuitive name for the task list
* Renamed the variable "idx" to "taskIdx" to make it clearer that it represents the index of a task
* Renamed the variable "by" to "endTime" to better convey its purpose as the end time of an event task

This enhances code maintainability and readability.

geraldngjx committed:
Merge pull request #2 from geraldngjx/branch-A-CodeQuality

Refactor code to enhance code quality

geraldngjx committed:
Add gradle.yml file for continuous integration

geraldngjx committed:
Add contact management feature to support add, remove and view contacts

GeraldBot did not support contact management previously.

Changes made:
* Enable adding new contacts into contact list.
* Enable removing existing contacts from contact list.
* Enable displaying existing contact list.

This feature enables contact management for GeraldBot.

geraldngjx committed:
Update gradle.yml file

geraldngjx committed:
Fix checkstyle issue by adding a whitespace

geraldngjx committed:
Merge pull request #3 from geraldngjx/branch-BCD-Extension

Add contact management feature to support add, remove and view contacts

geraldngjx committed:
Refactor code to inherit from parent classes to adhere to OOP

geraldngjx committed:
Fix CheckStyle issue

geraldngjx committed:
Add user guide and screenshot for website

geraldngjx committed:
Edit README file

geraldngjx committed:
Edit application description for the user guide
